undertow/src/main/java/org/wildfly/extension/undertow/security/jaspi/JASPICSecureResponseHandler.java

```diff
@@ -30,7 +30,7 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
                 ServletRequestContext requestContext = exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
                 String applicationIdentifier = JASPICAuthenticationMechanism.buildApplicationIdentifier(requestContext);
 
-                if (!JASPICAuthenticationMechanism.wasAuthExceptionThrown(exchange)) {
+                if (!JASPICAuthenticationMechanism.wasAuthExceptionThrown(exchange) && context != null) {
```

Comment:
Lines 30 and 31 can be moved inside this block, as the work done there has no side effects and the results are not used outside this block.

Reply:
I would also suggest to check first if context is not null and then check if an exception was thrown.

Reply:
@rjmveloso This is an old pull request that was merged over two years ago, realistically it is unlikely comments would be acted on.

We have subsequently added a WildFly Elytron JASPI implementation which will supersede the PicketBox integration, however if there are still changes needed in this implementation a Jira issue and PR would be a great help.